### ISSUE TITLE
crypto/stack: inconsistent semantics in OPENSSL_sk_push

### ISSUE THREAD

**Issue by void0red** [openssl/openssl#22462]
This bug was found when we were doing SFI fuzzing on openssl.

OPENSSL_sk_push return -1 when st is NULL
https://github.com/openssl/openssl/blob/b6eb95fa4439ea6254a5330487dabb2a499fb6c8/crypto/stack/stack.c#L397-L402
but OPENSSL_sk_insert return 0 when st is NULL
https://github.com/openssl/openssl/blob/b6eb95fa4439ea6254a5330487dabb2a499fb6c8/crypto/stack/stack.c#L258-L263

This will cause some caller checks to fail.
e.g. error handling doesn't cover the case where pol->qualifiers is null
https://github.com/openssl/openssl/blob/b6eb95fa4439ea6254a5330487dabb2a499fb6c8/crypto/x509/v3_cpols.c#L191-L200

**Comment by paulidale** [comment]
Yeah, our APIs aren't exactly consistent in their error return codes.  
This is documented, so it's not a bug.  It would be nice to be able to address things like this at some stage and the documentation could be improved.

**Comment by t8m** [comment]
@paulidale Some of the callers of `sk.*push()` functions treat -1 as success and they should not. That is IMO a bug.
